playground/src/app.jsx

```diff
@@ -420,15 +420,36 @@ const compile = (language, code) => {
     }
     if (problems) {
       return {
+        typeHints: [],
         problems: problems,
       }
     } else {
       return {
+        typeHints: compilation.type_hints.sort((a, b) => {
```

Comment:
I believe sort mutates the array and returns the reference to the same array. You might want to be immutable here with `.slice().sort()`

Reply:
yeah... i was aware but because we get a new hints array every time we recompile i was lazy and left the mutating version.